homework-solutions/js/hw-4/task-2.js

```diff
@@ -14,3 +14,275 @@
   5. Проверить задание со следующими значениями в переменной age: 10, 17, 18, 19, 59, 60, 61
 
   */
+
+
+  // 1. Создать переменную "minAge" и присвоить ей значение 18
+
+  // let minAge = 15;
+
+  // 2. Создать переменную "maxAge" и присвоить ей значение 60
+
+  // let maxAge = 60;
+
+  // 3. Создать переменную "age", в которую вы будете вводить возраст при выполнении программы
+
+  // let age;
+
+  // 4. Добавьте проверку: если тип данных в переменной age не number - вывести в консоль Incorrect data type
+
+  // if (typeof (age) !== "number"){
+  //   console.log("Incorrect data type")
+  // }
+
+  // 5. Создать if в котором будете проверять весь код переменной age со следующими условиями:
+  //   - Если age меньше чем minAge, вывести в консоль "You don't have access cause your age is " + "age" + " It's less then "
+  //   - Если "age" больше либо равно  minAge и меньше  maxAge, вывести в консоль "Welcome  !"
+  //   - Если "age" больше maxAge, вывести в консоль "Keep calm and look Culture channel".
+  //   - Иначе выводите "Technical work".
+
+  // if (typeof (age) !== "number"){
+  //   console.log("Incorrect data type")
+  // }
+  // else if (age < minAge){
+  //   console.log("You don't have access cause your age is " + age + " It's less then " + minAge)
+  // }
+  // else if (age >= minAge && age < maxAge){
+  //   console.log("Welcome  !")
+  // }
+  // else if (age > maxAge){
+  //   console.log("Keep calm and look Culture channel")
+  // }
+  // else {
+  //   console.log("Technical work")
+  // }
+
+  // 6. Проверить задание со следующими значениями в переменной age: 10, 17, 18, 19, 59, 60, 61
+
+// тут сам
+
+console.log("Age is 10:");
+
+  if(true){
+    let minAge = 15;
+
+  let maxAge = 60;
+
+  let age;
+
+  age = 10;
+
+  if (typeof (age) !== "number"){
+    console.log("Incorrect data type" + "\n---------------------------------\n");
+  }
+  else if (age < minAge){
+    console.log("You don't have access cause your age is " + age + " It's less then " + minAge + "\n---------------------------------\n");
+  }
```

Comment:
Не критично, но кажется так немного читабельнее вьіглядит ветвление
Вьіделили основную проверку и дальше, если прошли, прьігаем по другим
```suggestion
  if (typeof (age) !== "number"){
    console.log("Incorrect data type" + "\n---------------------------------\n");
  } else { 
     if (age < minAge){
    console.log("You don't have access cause your age is " + age + " It's less then " + minAge + "\n---------------------------------\n");
  }
```